### PR TITLE
Refactor and replace the HTTP client

### PR DESCRIPTION
**Pull request by kianmeng**
List of changes:

- Switch HTTP client from HTTPoison to Tesla
- Use concise way to handle URL params
- Fix and update ExVCR config, test cases, and cached result
- Remove all unused config by environment
- Follow the Tesla formatter rules
- Filter sensitive API token from cached HTTP response
- Downgrade the requirement from Elixir 1.9+ to 1.8+

### CODE REVIEW

**Comment by kianmeng** [comment]
@TORIFUKUKaiou Please help to test. :-)

File: lib/docomo_text_to_speech.ex [hunk 1]
```diff
@@ -4,65 +4,104 @@ defmodule DocomoTextToSpeech do
              |> String.split("<!-- MDOC !-->")
              |> Enum.fetch!(1)
 
+  use Tesla, only: [:post], docs: false
+
+  adapter Tesla.Adapter.Hackney
+  plug Tesla.Middleware.BaseUrl, "https://api.apigw.smt.docomo.ne.jp/futureVoiceCrayon/v1"
+  plug Tesla.Middleware.Query, API_KEY: api_key()
```
Comment (TORIFUKUKaiou) [inline lib/docomo_text_to_speech.ex]:
```suggestion
  plug Tesla.Middleware.Query, APIKEY: api_key()
```

File: lib/docomo_text_to_speech.ex [hunk 2]
```diff
@@ -4,65 +4,104 @@ defmodule DocomoTextToSpeech do
...
-    )
-    |> _format_current()
+  @spec run(String.t(), [conversion_option()]) :: {:ok | :error, any()}
+  def run(text, opts \\ []) do
+    params =
+      @default_opts
+      |> Keyword.merge(opts)
+      |> Keyword.put_new(:text_data, text)
+      |> Enum.into(%{}, fn {k, v} ->
+        {k |> to_string() |> Macro.camelize(), v}
```
Comment (TORIFUKUKaiou) [inline lib/docomo_text_to_speech.ex]:
```suggestion
        {k |> to_string() |> camelize(), v}
```

File: lib/docomo_text_to_speech.ex [hunk 3]
```diff
@@ -4,65 +4,104 @@ defmodule DocomoTextToSpeech do
...
+      Application.get_env(:docomo_text_to_speech, :api_key)
+  end
+
+  defp handle_response({:error, _} = error), do: error
+  defp handle_response({:ok, %Tesla.Env{status: 200, body: body}}), do: {:ok, body}
+
+  defp handle_response({:ok, %Tesla.Env{status: status, body: body}}) do
+    %{"requestError" => %{"policyException" => %{"messageId" => msg_id, "text" => msg}}} = body
+    {:error, "(#{status}) #{msg_id}: #{msg}"}
+  end
```
Comment (TORIFUKUKaiou) [inline lib/docomo_text_to_speech.ex:120]:
```suggestion
  end
  
  defp camelize("speaker_id"), do: "SpeakerID"

  defp camelize("style_id"), do: "StyleID"

  defp camelize(key), do: Macro.camelize(key)
```

Macro.camelize("speaker_id") == "SpeakerId", but api server wants "SpeakerID".
Macro.camelize("style_id") == "SpeakerId", but api server wants "StyleID".

File: mix.exs [hunk 4]
```diff
@@ -27,11 +28,12 @@ defmodule DocomoTextToSpeech.MixProject do
 
   defp deps do
     [
-      {:httpoison, "~> 1.6"},
+      {:hackney, "~> 1.15.2"},
```
Comment (TORIFUKUKaiou) [inline mix.exs]:
```suggestion
      {:hackney, "~> 1.16"},
```

hackney 1.15.2 causes the below.
```
** (ArgumentError) raise/1 and reraise/2 expect a module name, string or exception as the first argument, got: {:options, {:sslv3, {:versions, [:"tlsv1.2", :"tlsv1.1", :tlsv1, :sslv3]}}}
    (docomo_text_to_speech 0.2.1) lib/docomo_text_to_speech.ex:91: DocomoTextToSpeech.run!/2
```

File: lib/docomo_text_to_speech.ex [hunk 5]
```diff
@@ -4,65 +4,104 @@ defmodule DocomoTextToSpeech do
...
-      speaker_id,
-      style_id,
-      speech_rate,
-      power_rate,
-      voice_type,
-      audio_file_format
-    )
-    |> _format_current()
+  @spec run(String.t(), [conversion_option()]) :: {:ok | :error, any()}
+  def run(text, opts \\ []) do
```
Comment (TORIFUKUKaiou) [inline lib/docomo_text_to_speech.ex:63]: 👍 
Thank you.
Keyword List!

**Review by TORIFUKUKaiou — COMMENTED** [review]
Thank you for refactoring.

I test it.
I modify a little to get a right response.
Would you please see my comments?

I don't know Tesla.
What is Tesla better than HTTPoison?

File: mix.exs (hunk 4 above)
Comment (kianmeng) [reply mix.exs]: > hackney 1.15.2 causes the below.
> 
> ```
> ** (ArgumentError) raise/1 and reraise/2 expect a module name, string or exception as the first argument, got: {:options, {:sslv3, {:versions, [:"tlsv1.2", :"tlsv1.1", :tlsv1, :sslv3]}}}
>     (docomo_text_to_speech 0.2.1) lib/docomo_text_to_speech.ex:91: DocomoTextToSpeech.run!/2
> ```

What the value of your `.tool-versions` or rather, what Elixir and Erlang OTP version you're running?

**Comment by kianmeng** [comment]
> Thank you for refactoring.
> 
> I test it.
> I modify a little to get a right response.
> Would you please see my comments?

Done. Please check and test again.

> I don't know Tesla.
> What is Tesla better than HTTPoison?

Code is simpler, especially with middlewares.

File: mix.exs (hunk 4 above)
Comment (TORIFUKUKaiou) [reply mix.exs]: I use elixir 1.10.4-otp-23 and erlang 23.0.1 .

**Review by TORIFUKUKaiou — APPROVED** [review]
Thanks for the refactoring. I test it. It can get right response from the API. 
Sorry for the late response.

**Pull request merged**